src/main/java/com/rufus/bumblebee/controllers/ContainerController.java

```diff
@@ -21,21 +23,33 @@
 public class ContainerController {
 
     private final ContainerService service;
+    private final HistoryService<HistoryDto, String> historyService;
 
     @Autowired
-    public ContainerController(ContainerService service) {
+    public ContainerController(ContainerService service, HistoryService<HistoryDto, String> historyService) {
         this.service = service;
+        this.historyService = historyService;
     }
 
-    @PostMapping(path = "/add")
-    public ResponseEntity<ContainerDto> addContainer(@RequestBody ContainerRequest request) {
-        return new ResponseEntity<ContainerDto>(
-                service.createContainer(request.getName(), request.getAuth(), request.getReportType()), HttpStatus.OK
+    @PostMapping(path = "/add", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<ContainerDto> addContainer(@RequestBody ContainerRequest request) throws Exception {
+        return new ResponseEntity<>(
+                service.createContainer(request.getName(), request.getHistoryOn(), request.getReportType()), HttpStatus.OK
         );
     }
 
-    @DeleteMapping(path = "/remove/{cuid}")
-    public ResponseEntity<String> removeContainer(@PathVariable("cuid") @NotEmpty String cuid) throws NoResultException {
-        return new ResponseEntity<String>("The container with the ID " + service.removeContainer(cuid) + " was deleted", HttpStatus.OK);
+    @DeleteMapping(path = "/remove/{cuid}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> removeContainer(@PathVariable("cuid") @NotEmpty String cuid) throws Exception {
+        return new ResponseEntity<>("The container with the ID " + service.removeContainer(cuid) + " was deleted", HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/history/{cuid}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<HistoryDto> getHistory(@PathVariable("cuid") @NotEmpty String cuid){
+        return new ResponseEntity<>(historyService.getHistory(cuid), HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/information/{name}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<ContainerDto> getContainerByName(@PathVariable("name") @NotEmpty String name) throws Exception {
+        return new ResponseEntity<>(service.getContainerByName(name), HttpStatus.OK);
```

Comment:
JFYI, Вы можете также использовать статический метод:
```suggestion
        return ResponseEntity.ok(service.getContainerByName(name));
```